homeworks/vlad.makedon_Vladmaked/homework_6/user-info.js

```diff
@@ -0,0 +1,29 @@
+const user = {};
+
+Object.defineProperties(user, {
+    fullName: {
+        get: () => this.fullName,
+        set: (value) => {
+            this.fullName = value.split('').reduce((accumulator, element, index, array) => (
+                index === 0 || array[index - 1] === ' ' || array[index - 1] === '-'
+                    ? accumulator + element.toUpperCase()
+                    : accumulator + element.toLowerCase()
+            ), '');
+        },
+    },
+    phone: {
+        get: () => this.phone,
+        set: (value) => {
+            this.phone = value.split('').reduce((accumulator, element, index) => (
+                (index === 0 && element === '+') || !Number.isNaN(parseInt(element, 10))
+                    ? accumulator + element
```

Comment:
here as well, RegExp is suitable solution for this kind of task... just split string by all non-digit characters